wrappers/src/fdw/s3_fdw/parquet.rs

```
@@ -63,7 +63,7 @@ impl S3ParquetReader {
         )
         .map_err(to_io_error)
         .map(|output| {
-            let object_size = output.object_size() as u64;
+            let object_size = output.object_size().unwrap_or_default() as u64;
```

if `output.object_size()` returns `None`, `object_size` will be zero. Is that ok? I'm not too familiar with this code, maybe @burmecia can comment.

---

we're assuming `object_size` always available even it is zero, so should return an error here. I've made the change.

---

Thanks!